src/hotspot/share/gc/g1/g1GCPhaseTimes.cpp

```diff
@@ -519,8 +519,6 @@ void G1GCPhaseTimes::print_other(double accounted_ms) const {
 }
 
 void G1GCPhaseTimes::print() {
-  note_gc_end();
-
   if (_cur_verify_before_time_ms > 0.0) {
```

Comment:
Is this `if` just checking whether verification is enabled or not? If so, can `VerifyBeforeGC` be used? Otherwise, it's unclear to me why we are comparing it with `0.0`.

Reply:
The `if` checks whether any verification work has been done (it will be > 0.0 if it has been set) in this pause. We can't just use `VerifyBefore/AfterGC` because verification before/after does not always run for all young gc pauses (see `VerifyGCType`).
If we want to change this (existing) behavior to like always print the time if the flag is enabled I would suggest to file another CR instead of hiding it here (I can do all that and the associated work if that is desired).

Reply:
> We can't just use VerifyBefore/AfterGC because...

I see; thank you for the explanation.

> If we want to change this (existing) behavior to like always print the time if the flag is enabled

Either that or just put some comment there, since in what situation the condition is true is not obvious. Ofc, this is very subjective.